testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/k8s/HyperfoilKubernetesOperatorProvisionerTest.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.intersmash.testsuite.provision.k8s;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import org.jboss.intersmash.application.operator.HyperfoilOperatorApplication;
+import org.jboss.intersmash.junit5.IntersmashExtension;
+import org.jboss.intersmash.k8s.KubernetesConfig;
+import org.jboss.intersmash.k8s.client.Kuberneteses;
+import org.jboss.intersmash.provision.k8s.HyperfoilKubernetesOperatorProvisioner;
+import org.jboss.intersmash.provision.olm.OperatorGroup;
+import org.jboss.intersmash.provision.operator.HyperfoilOperatorProvisioner;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.runschema.RunStatisticsWrapper;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.HyperfoilApi;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.ApiClient;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.ApiException;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.Configuration;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.model.Run;
+import org.jboss.intersmash.testsuite.junit5.categories.KubernetesTest;
+import org.jboss.intersmash.testsuite.k8s.NamespaceCreationCapable;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.hyperfoil.v1alpha2.Hyperfoil;
+import io.hyperfoil.v1alpha2.HyperfoilBuilder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@KubernetesTest
+public class HyperfoilKubernetesOperatorProvisionerTest implements NamespaceCreationCapable {
```

Comment:
couldn't we use a single interface and deduce if K8 OR OCP from some context? I am assuming we run against K8 OR OCP, not the two at the same time in the same test run

Reply:
Do you mean for `Project` vs. `Namespace` creation?
Tests are meant to be different, depending on whether they would target Kubernetes or OpenShift, and such detail (i.e. whether to create a Namespace or a Project), is one of the details that should be set explicitly to decouple the provisioning functionality from the execution context.
The same applies if you're relating to K8s vs. OpenShift tests. In fact, it is very likely that different resources or processes would be verified, depending on the target platform.
Of course there could also be the chance that K8s or OpenShift scenarios and tests could be similar, but this should not break the original requirement to be able and deploy/tests on both the platforms.

IIUC what you are relating to, is the ability for a test suite that uses Intersmash to execute a set of tests _just_ against K8s or OpenShift in one  single run. This is something that must be implemented in the domain of the client test suite, and not by Intersmash. We actually do something similar in the Intersmash testsuite, where we use the `OpenShiftTest` or `KubernetesTest` categories to select which tests should be run.